actors/evm/src/interpreter/system.rs

```diff
@@ -91,6 +91,11 @@ pub struct System<'r, RT: Runtime> {
     bytecode: Option<EvmBytecode>,
     /// The contract's EVM storage slots.
     slots: StateKamt<RT::Blockstore>,
+
+    /// The contract's EVM transient storage slots.
+    transient_slots: StateKamt<RT::Blockstore>,
+    current_transient_data_lifespan: Option<TransientDataLifespan>,
```

Comment:
```suggestion
    current_transient_data_lifespan: TransientDataLifespan,
```

There's no need for this to be an option, I think? If we fail to load state information, abort; that means there's a bug somewhere.

Reply:
Yes! I agree - thanks - looking into it

Reply:
One concern comes from get_current_transient_data_lifespan returning None - which happens at least in the test context - I'm looking into what happens if I throw an error if get_current_transient_data_lifespan returns None in load/new and see how that effects the tests

Reply:
I'm not sure how to handle current_transient_data_lifespan, in [new()](https://github.com/filecoin-project/builtin-actors/blob/3d5deba2a526d07fa26ab0eb5ed3ff3a3c9fd62e/actors/evm/src/interpreter/system.rs#L113) other than to allow current_transient_data_lifespan to be None. I could use a default value but I think None would be more accurate

Reply:
Also any reason for [tombstone to be an option](https://github.com/filecoin-project/builtin-actors/blob/3d5deba2a526d07fa26ab0eb5ed3ff3a3c9fd62e/actors/evm/src/interpreter/system.rs#L110) should apply to current_transient_data_lifespan. I think reverting to keeping current_transient_data_lifespan a option makes the most sense for now. I can't make sense of how to implement `new()` or `load()` to handle get_current_transient_data_lifespan returning None without making current_transient_data_lifespan an option. LMK if that makes sense

Reply:
Thanks for explaining on zoom - I updated the changes you requested. However there is a single failing test. [test_origin_ethaddr ](https://github.com/filecoin-project/builtin-actors/blob/837a003ab61a92338afec75245dc35b0b84593d7/actors/evm/src/interpreter/instructions/context.rs#L368) fails because it does not have an id associated with it.

Reply:
I think this fixes it - 3c6d9a7d - I'm changing the origin from type id/u64 to type Address

Reply:
The origin is always an ID address. I'd just remove that test entirely (I'm not sure why it's there).

Reply:
great thanks for clarifying - i reverted the commit that switched from id to address and deleted the test 037939aae6db470d3d9b18c3623a001ce0172977 and dddf19c820ff27b6877b28ba157bfc7e480d361f